tests/CommonDatatypeTest.php

```diff
@@ -34,4 +34,72 @@ public function testToArray()
         $datatype = new Common("VARCHAR", "50", true);
         $this->assertEquals(["type" => "VARCHAR", "length" => "50", "nullable" => true], $datatype->toArray());
     }
+
+    public function testToMetadata()
+    {
+        $datatype = new Common("VARCHAR");
+
+        $md = $datatype->toMetadata();
+        foreach ($md as $mdat) {
+            $this->assertArrayHasKey('key', $mdat);
+            $this->assertArrayHasKey('value', $mdat);
+            if ($mdat['key'] === "KBC.datatype.type") {
+                $this->assertEquals('VARCHAR', $mdat['value']);
+            } else if ($mdat['key'] === 'KBC.datatype.length') {
+                $this->fail("unspecified length should not create metadata.");
+            } else if ($mdat['key'] === 'KBC.datatype.nullable') {
+                $this->assertEquals(true, $mdat['value']);
+            } else if ($mdat['key'] === 'KBC.datatype.format') {
+                $this->fail("unspecified format should not create metadata.");
+            } else if ($mdat['key'] === 'KBC.datatype.default') {
+                $this->assertEquals("NULL", $mdat['value']);
+            } else if ($mdat['key'] === 'KBC.datatype.basetype') {
+                $this->assertEquals('STRING', $mdat['value']);
+            }
+        }
+
+        $datatype = new Common("DATE", "10", false, "1970-01-01", "Y-m-d");
+        $md = $datatype->toMetadata();
+        foreach ($md as $mdat) {
+            $this->assertArrayHasKey('key', $mdat);
+            $this->assertArrayHasKey('value', $mdat);
+            if ($mdat['key'] === "KBC.datatype.type") {
+                $this->assertEquals('DATE', $mdat['value']);
+            } else if ($mdat['key'] === 'KBC.datatype.length') {
+                $this->assertEquals("10", $mdat['value']);
+            } else if ($mdat['key'] === 'KBC.datatype.nullable') {
+                $this->assertEquals(false, $mdat['value']);
+            } else if ($mdat['key'] === 'KBC.datatype.format') {
+                $this->assertEquals("Y-m-d", $mdat['value']);
+            } else if ($mdat['key'] === 'KBC.datatype.default') {
+                $this->assertEquals("1970-01-01", $mdat['value']);
+            } else if ($mdat['key'] === 'KBC.datatype.basetype') {
+                $this->assertEquals('DATE', $mdat['value']);
+            }
+        }
+    }
+
+    public function testGetBasetype()
+    {
+        $this->assertEquals("DATE", (new Common("date"))->getBasetype());
+        $this->assertEquals("DATE", (new Common("DATE"))->getBasetype());
+        $this->assertEquals("TIMESTAMP", (new Common("DateTime"))->getBasetype());
+        $this->assertEquals("TIMESTAMP", (new Common("timestamp"))->getBasetype());
+        $this->assertEquals("STRING", (new Common("dattim"))->getBasetype());
```

Comment:
no, that's why it should return "STRING" as that's catchall.  it's a bit of a pointless test since the `anythingelse` test at the bottom covers that.